heron/packing/src/java/com/twitter/heron/packing/roundrobin/ResourceCompliantRRPacking.java

```diff
@@ -153,7 +163,45 @@ public PackingPlan pack() {
     Map<String, Long> ramMap = TopologyUtils.getComponentRamMapConfig(topology);
 
     Set<PackingPlan.ContainerPlan> containerPlans = PackingUtils.buildContainerPlans(
-        containerInstances, ramMap, this.defaultInstanceResources, paddingPercentage);
+        resourceCompliantRRAllocation.get(), ramMap, this.defaultInstanceResources,
+        paddingPercentage);
+    return new PackingPlan(topology.getId(), containerPlans);
+  }
+
+  /**
+   * Get a new packing plan given an existing packing plan and component-level changes.
+   *
+   * @return new packing plan
+   */
+  public PackingPlan repack(PackingPlan currentPackingPlan, Map<String, Integer> componentChanges) {
+    int adjustments = 0;
+    this.numAdjustments = 0;
+    this.numContainers = currentPackingPlan.getContainers().size();
+    this.containerId = 1;
+
+    int additionalContainers = computeNumAdditionalContainers(componentChanges, currentPackingPlan);
+    increaseNumContainers(additionalContainers);
```

Comment:
Sorry, I missed that `increaseNumContainers` doesn't change `numAdjustments` and `containerId`, only `numContainers` so I was confusing things by lumping that method in with the reset I was suggesting. I'm suggesting a method that resets the following, which happen in multiple places:
```
this.numAdjustments = 0;
this.containerId = 1;
```